### DISCUSSION TITLE
ipv8 based linux package management

### DISCUSSION THREAD

**Discussion by hbiyik** [Tribler/tribler#7292] [Ideas]
i think if the scalability aspect of ip-v8 could allow, it would be a very interesting to have a linux packaging + distribution system using ipv8. The source codes could be distributed to users, and compiled on the users end much like gentoo or arch. and users would seed the built binaries to users again.

Yet, there would definitely a  need for an at least semi central notary system for security, i think a form of distributed trust based on ranking would be an invitation for the botnets.

Just a weird idea popped up in my mind.

**Comment by nitmeare** [comment]
Potentially one could take all the Debian/Ubuntu .deb files from the official repository and seed them in Tribler as torrent. Then make an addon to apt-get so it gets the Magnet Links and downloads the packages using tribler network instead through FTP/HTTP. I think with some programming it would be possible to convert any existing distribution towards using a decentralized encrypted package management.

The .deb files then would still be secured by the Distributions GPG Hashing. So there would be no need to set up any "semi central notary system" as you call it cause the validation would still be provided by the original distributor.

**Reply by hbiyik** [reply]
ah makes sense, sometimes solution is much simpler :). And in this case you even need tribler just a torrent client. I wonder if any linux distro has ever tried this.

I think having all the packages is not the best practise because package sizes would be so small that initializing a torrent would take so much than actually downloading. May be a snapshot mechanism can be considered like bunch of small packages bundled in some snapshot and revisions of the snapshot is seeded. However small sub packages revisions of snapshot would introduce factorial times of the actual big snapshot revision. Hmm...
